packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_impl.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:amplify_auth_cognito/src/credentials/legacy_credential_provider_android.dart';
+import 'package:amplify_auth_cognito/src/credentials/legacy_credential_provider_ios.dart';
+import 'package:amplify_auth_cognito_dart/amplify_auth_cognito_dart.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_auth_cognito.legacy_credential_provider_impl}
+/// The implementation of [LegacyCredentialProvider] for migrating
+/// credentials from the legacy iOS and Android SDKs.
+/// {@endtemplate}
+class LegacyCredentialProviderImpl implements LegacyCredentialProvider {
+  /// {@macro amplify_auth_cognito.legacy_credential_provider_impl}
+  LegacyCredentialProviderImpl(CognitoAuthStateMachine stateMachine)
+      : _stateMachine = stateMachine;
+  final CognitoAuthStateMachine _stateMachine;
+
+  late final LegacyCredentialProvider? _instance = () {
```

Comment:
Since this only gets instantiated in `auth_plugin_impl` in a method where we've already checked if we're on iOS/Android, I think you can safely remove all the platform-related null checks since it should be impossible to get here otherwise.